### PR TITLE
(CDAP-1993) Allow Spark program setting SparkConf to be used during e…

### PR DESCRIPTION
**Pull request by chtyim**
…xecution.

### CODE REVIEW

**Comment by rohitsinha54** [comment]
LGTM

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkRuntimeService.java [hunk 1]
```diff
@@ -299,13 +300,18 @@ public Void call() throws Exception {
...
     ImmutableList.Builder<String> builder = ImmutableList.builder();
 
+    // Add user specified configs first. CDAP specifics config will override them later if there are duplicates.
+    for (Tuple2<String, String> tuple : context.getSparkConf().getAll()) {
+      builder.add("--conf").add(tuple._1() + "=" + tuple._2());
+    }
+
     builder.add("--class").add(SparkProgramWrapper.class.getName())
            .add("--master").add(executionMode)
-           .add("--deploy-mode").add("client")
```
Comment (rohitsinha54) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkRuntimeService.java]: The client deploy mode is not required (a limitation) anymore ?

Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkRuntimeService.java]: See https://github.com/caskdata/cdap/pull/3162/files#diff-a5929eb4d21c9863e4b08a96b8e3744fR43

Comment (rohitsinha54) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkRuntimeService.java]: Cool.

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-DUT2230-1

**Pull request merged**